skoolkit/basic.py

```diff
@@ -114,14 +115,92 @@
     255: 'COPY'
 }
 
+def get_number(snapshot, i):
+    if snapshot[i] == 0:
+        # Small integer (signed)
+        num = _get_integer(snapshot, i)
+    else:
+        # Floating point number (signed)
+        num = _get_float(snapshot, i)
+    return num
+
+def _get_integer(snapshot, i):
```

Comment:
This method need not be so long and complicated. Let's assume for now that the sign byte is either 0 (positive) or 255 (negative). Try this:

    if snapshot[i + 1]:
        return get_word(snapshot, i + 2) - 65536
    return get_word(snapshot, i + 2)